cmake/modules/AddSwift.cmake

```diff
@@ -997,7 +1021,7 @@ function(_add_swift_library_single target name)
     set_target_properties("${target}"
       PROPERTIES
       INSTALL_NAME_DIR "${install_name_dir}")
-  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" AND NOT "${SWIFTLIB_SINGLE_SDK}" STREQUAL "ANDROID")
```

Comment:
I may be misunderstanding the question but if you're referring to swift's dynamic library dependencies on android, they currently need to be added to the android project by hand alongside libswiftCore.so.

Then you use java's dlib api, i.e. `System.loadLibrary("swiftCore")`